hyperapp.d.ts

```diff
@@ -54,56 +56,58 @@ export type ActionResult<State extends Hyperapp.State> =
   | null
   | void
 
-export type ActionCallers<
-  State extends Hyperapp.State,
-  Actions extends Hyperapp.Actions<State>
-> = {
-  [P in keyof Actions]: (
-    data: any
-  ) => ActionResult<State> | ActionCallers<State, Actions>
-}
 
-export interface Actions<State extends Hyperapp.State> {
+export interface Actions {
   [action: string]:
-    | Actions<State>
+    | Actions
     | ((
-        state: State,
-        actions: ActionCallers<State, Actions<State>>,
         data: any
       ) => ActionResult<State>)
 }
 
+export type ActionsImpl<
```

Comment:
As @JorgeBucaran suggested, a more pleasant name would be nice. Though (if I understand this PR correctly) since this is merely and internal type, it may not be necessary.

Reply:
Actually, do we need to export this type at all? I believe this type is only used by the `App` interface to define the structure of the `actions` option prop and not intended to be directly extended by users. @Mytrill, is that correct?

Reply:
You will need it to implement the actions:

```typescript
const actions. ActionsImpl<State, Actions> = {...}
```

I will add it in index.tsx.

Reply:
Oh good call - that makes sense.